### PR TITLE
[AIR/Serve] Auto-enable GPU Prediction

### PR DESCRIPTION
**Pull request by amogkam**
Automatically enable GPU prediction for Predictors if num_gpus is set for the PredictorDeployment

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/BUILD [hunk 1]
```diff
@@ -339,6 +339,14 @@ py_test(
     deps = [":serve_lib"],
 )
 
+py_test(
+    name = "test_air_integrations",
```
Comment (simon-mo) [inline python/ray/serve/BUILD:343]: do we need a new name here?

File: python/ray/serve/air_integrations.py [hunk 2]
```diff
@@ -19,6 +21,8 @@
 except ImportError:
     pd = None
 
+logger = logging.getLogger(__name__)
```
Comment (simon-mo) [inline python/ray/serve/air_integrations.py:24]:
```
from ray.serve.constants import SERVE_LOGGER_NAME

logger = logging.getLogger(SERVE_LOGGER_NAME)
```

File: python/ray/serve/BUILD (hunk 1 above)
Comment (amogkam) [inline python/ray/serve/BUILD:343]:
```suggestion
    name = "test_air_integrations_gpu",
```

Comment (amogkam) [reply python/ray/serve/BUILD]: Updated!

File: python/ray/serve/air_integrations.py (hunk 2 above)
Comment (amogkam) [reply python/ray/serve/air_integrations.py]: Updated!

**Comment by jiaodong** [comment]
content lgtm, will check later once CI passed

**Comment by amogkam** [comment]
@jiaodong could you take a look pls when you get a chance if you would like? that way we can merge once tests pass

**Review by jiaodong — APPROVED** [review]
i didn't have any issue with the code here so mostly just discretion to ensure new CI tests get triggered and pass. I will approve for now to unblock so you can merge once you've verified CI later

File: .buildkite/pipeline.gpu.yml [hunk 3]
```diff
@@ -20,6 +20,21 @@
     - bazel test --config=ci $(./ci/run/bazel_export_options) --build_tests_only --jobs 1
       --test_tag_filters=gpu --test_env=RAY_USE_MULTIPROCESSING_CPU_COUNT=1 --test_env=RLLIB_NUM_GPUS=1 rllib/...
 
+
+- label: ":tv: :serverless: Serve Tests"
+  parallelism: 3
```
Comment (simon-mo) [inline .buildkite/pipeline.gpu.yml:25]:
```suggestion
```

just a single test, no need for parallel builder

**Review by simon-mo — APPROVED** [review]

**Comment by scv119** [comment]
is it still to be picked? no activity for 12 days

**Pull request merged**